stan/math/rev/functor/coupled_ode_system.hpp

```diff
@@ -119,7 +119,10 @@ struct coupled_ode_system<F, double, var> {
     try {
       start_nested();
 
-      const vector<var> y_vars(z.begin(), z.begin() + N_);
+      vector<var> y_vars;
+      y_vars.reserve(N_);
+      for (std::size_t i = 0; i < N_; ++i)
+        y_vars.emplace_back(var(new vari(z[i], false)));
```

Comment:
This should use a range-based for-loop and a simplifed emplace-back:

```
for (double zi : z)
  y_vars.emplace_back(new vari(zi, false)));
```

The reason to simplify `emplace_back` is that it is able to takes the arguments of a consructor for `var` and do the construction in place.

Reply:
Thanks. Fixed. For this one and for the placement of the theta's on the outer stack as well.